### PR TITLE
Document LazyString in performance tips

### PR DESCRIPTION
**Pull request by lassepe**
Does what it says. Doc-changes only.

Recent motivating example: https://github.com/FluxML/Flux.jl/issues/2399

### CODE REVIEW

File: doc/src/manual/performance-tips.md [hunk 1]
```diff
@@ -1239,6 +1239,34 @@ versus:
 println(file, f(a), f(b))
 ```
 
+## Avoid eager string materialization
+
+In settings where a string representation of an object is only needed
+conditionally (e.g. in error paths of functions or conditional warnings such as
+deprecations), it is advisable to avoid the overhead of eagerly materializing
+the string. Since Julia 1.8, this can be achieved with via
```
Comment (Zentrik) [inline doc/src/manual/performance-tips.md:1247]: I think you can remove either 'with' or 'via'.

Comment (lassepe) [reply doc/src/manual/performance-tips.md]: Good catch!

Comment (lassepe) [reply doc/src/manual/performance-tips.md]: Fixed

**Review by aviatesk — APPROVED** [review]
Most of the performance boosts usually come from compiler optimizations, thanks to the improved effects by using `LazyString`. But, details this internal might not be necessary for the performance tips section.

**Comment by lassepe** [comment]
@aviatesk those are interesting additional details. But I think they are best added to the docstring of `LazyString` itself to keep the performance tips brief.

**Comment by aviatesk** [comment]
Yep, it's fine.

**Comment by lassepe** [comment]
Is there anything left to do before this can be merged?

**Comment by aviatesk** [comment]
Could you fix the CI build failure? It seems to be caused by this PR.

**Comment by lassepe** [comment]
All done. I think the remaining breakage is unrelated.

**Pull request merged**